backend/app/models.py

```diff
@@ -40,15 +42,58 @@ class TimeStampMixin(SQLModel):
     )
 
 
-class GerryDBTable(TimeStampMixin, SQLModel, table=True):
+class DistrictrMap(TimeStampMixin, SQLModel, table=True):
     uuid: str = Field(sa_column=Column(UUIDType, unique=True, primary_key=True))
-    name: str = Field(nullable=False, unique=True)
+    name: str = Field(nullable=False)
+    # This is intentionally not a foreign key on `GerryDBTable` because in some cases
+    # this may be the GerryDBTable but in others the pop table may be a materialized
+    # view of two GerryDBTables in the case of shatterable maps.
+    # We'll want to enforce the constraint tha the gerrydb_table_name is either in
+    # GerrydbTable.name or a materialized view of two GerryDBTables some other way.
+    gerrydb_table_name: str | None = Field(nullable=True, unique=True)
+    # Null means default number of districts? Should we have a sensible default?
+    num_districts: int | None = Field(nullable=True, default=None)
     tiles_s3_path: str | None = Field(nullable=True)
+    parent_layer: str = Field(
+        sa_column=Column(UUIDType, ForeignKey("gerrydbtable.uuid"), nullable=False)
+    )
+    child_layer: str | None = Field(
+        sa_column=Column(
+            UUIDType, ForeignKey("gerrydbtable.uuid"), default=None, nullable=True
+        )
+    )
+    # schema? will need to contrain the schema
+    # where does this go?
+    # when you create the view, pull the columns that you need
+    # we'll want discrete management steps
 
 
-class GerryDBViewPublic(BaseModel):
+class DistrictrMapPublic(BaseModel):
     name: str
-    tiles_s3_path: str | None
+    gerrydb_table_name: str
+    parent_layer: str
+    child_layer: str | None = None
+    tiles_s3_path: str | None = None
+    num_districts: int | None = None
+
+
+class GerryDBTable(TimeStampMixin, SQLModel, table=True):
+    uuid: str = Field(sa_column=Column(UUIDType, unique=True, primary_key=True))
+    # Must correspond to the layer name in the tileset
+    name: str = Field(nullable=False, unique=True)
+
+
+class ParentChildEdges(TimeStampMixin, SQLModel, table=True):
+    districtr_map: str = Field(
```

Comment:
(i) how do we feel about partitioning this on districtr map as well?

Reply:
Probs a good idea!

Reply:
done